pkg/sql/alter_column_type_test.go

```diff
@@ -0,0 +1,521 @@
+	var x, z int
+	dest := []interface{}{&x, &y, &z}
+
+	expected := [][]interface{}{{1, "1", 1}, {2, "2", 2}, {3, "HELLO", 3}}
+	err := sql.QueryExpect(`SELECT * FROM d.t3 ORDER BY x`, sqlDB, expected, dest)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestAlterColumnTypeBeforeAlterPrimaryKey(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	swapNotification := make(chan struct{})
+	waitBeforeContinuing := make(chan struct{})
+
+	params, _ := tests.CreateTestServerParams()
+	params.Knobs = base.TestingKnobs{
+		SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
+			RunBeforeComputedColumnSwap: func() {
+				swapNotification <- struct{}{}
+				<-waitBeforeContinuing
+			},
+		},
+		DistSQL: &execinfra.TestingKnobs{},
+	}
+
+	s, sqlDB, _ := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(ctx)
+
+	if _, err := sqlDB.Exec(
+		`SET enable_alter_column_type_general = true;`); err != nil {
+		t.Error(err)
+	}
+
+	if _, err := sqlDB.Exec(`
+CREATE DATABASE t;
+CREATE TABLE t.test (x INT NOT NULL);
+`); err != nil {
+		t.Error(err)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		if _, err := sqlDB.Exec(`
+ALTER TABLE t.test ALTER COLUMN x TYPE STRING;
+`); err != nil {
+			t.Error(err)
+		}
+		wg.Done()
+	}()
+
+	<-swapNotification
+
+	expected := "pq: unimplemented: table test is currently undergoing a schema change"
+	_, err := sqlDB.Exec(`ALTER TABLE t.test ALTER PRIMARY KEY USING COLUMNS (x);`)
+	waitBeforeContinuing <- struct{}{}
+	wg.Wait()
+	if err.Error() != expected {
+		t.Fatalf("expected error: %s, got: %s", expected, err.Error())
+	}
+}
+
+func TestAlterPrimaryKeyBeforeAlterColumnType(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	swapNotification := make(chan struct{})
+	waitBeforeContinuing := make(chan struct{})
+
+	params, _ := tests.CreateTestServerParams()
+	params.Knobs = base.TestingKnobs{
+		SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
+			RunBeforePrimaryKeySwap: func() {
+				swapNotification <- struct{}{}
+				<-waitBeforeContinuing
+			},
+		},
+		DistSQL: &execinfra.TestingKnobs{},
+	}
+
+	s, sqlDB, _ := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(ctx)
+
+	if _, err := sqlDB.Exec(`
+CREATE DATABASE t;
+CREATE TABLE t.test (x INT NOT NULL);
+`); err != nil {
+		t.Error(err)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		if _, err := sqlDB.Exec(`
+ALTER TABLE t.test ALTER PRIMARY KEY USING COLUMNS (x);
+`); err != nil {
+			t.Error(err)
+		}
+		wg.Done()
+	}()
+
+	<-swapNotification
+
+	expected := "pq: unimplemented: table test is currently undergoing a schema change"
+	_, err := sqlDB.Exec(`
+SET enable_alter_column_type_general = true;
+ALTER TABLE t.test ALTER COLUMN x TYPE STRING;`)
+	waitBeforeContinuing <- struct{}{}
+	wg.Wait()
+	if err.Error() != expected {
+		t.Fatalf("expected error: %s, got: %s", expected, err.Error())
+	}
+}
+
+func TestAlterColumnTypeWithAlterPrimaryKey(t *testing.T) {
```

Comment:
This test seems like it just depends on alter column working and then alter primary key working -- what does it add?